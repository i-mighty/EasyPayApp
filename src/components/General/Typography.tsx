import styled from 'styled-components/native';
import { Text } from 'react-native';

const TitleText = styled(Text)<{ color?: string }>`
  font-size: 21px;
  color: ${({ theme, color }) => (color ? color : theme.vars.black)};
`;

const ItalicsText = styled(Text)<{ color?: string }>`
  font-style: italic;
  color: ${({ theme, color }) => (color ? color : theme.vars.black)};
`;

const PlainText = styled(Text)<{ color?: string }>`
  font-style: normal;
  color: ${({ theme, color }) => (color ? color : theme.vars.black)};
`;
export { TitleText, ItalicsText, PlainText };
