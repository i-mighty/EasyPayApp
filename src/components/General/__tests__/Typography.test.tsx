import React from 'react';
import { renderWithProps } from '@src/utils/test-utils';
import { TitleText, ItalicsText, PlainText } from '../Typography';
import theme from '@src/utils/theme';

const titleDefaultProps = {
  testID: 'TitleText',
};
const titleColoredProps = {
  ...titleDefaultProps,
  color: theme.vars.orange,
};

const italicsDefaultProps = {
  testID: 'ItalicsText',
};
const italicsColoredProps = {
  ...italicsDefaultProps,
  color: theme.vars.orange,
};

const plainDefaultProps = {
  testID: 'PlainText',
};
const plainColoredProps = {
  ...plainDefaultProps,
  color: theme.vars.orange,
};

describe('TitleText', () => {
  it('renders properly', () => {
    const { getByTestId } = renderWithProps(
      <TitleText {...titleDefaultProps} />,
    );
    const textComponent = getByTestId(titleDefaultProps.testID);
    expect(textComponent).toHaveStyleRule('font-size', 21);
    expect(textComponent).toHaveStyleRule('color', theme.vars.black);
    expect(textComponent).toMatchSnapshot();
  });

  it('renders with the color from prop', () => {
    const { getByTestId } = renderWithProps(
      <TitleText {...titleColoredProps} />,
    );
    const textComponent = getByTestId(titleColoredProps.testID);
    expect(textComponent).toHaveStyleRule('font-size', 21);
    expect(textComponent).toHaveStyleRule('color', italicsColoredProps.color);
    expect(textComponent).toMatchSnapshot();
  });
});

describe('ItalicsText', () => {
  it('renders properly', () => {
    const { getByTestId } = renderWithProps(
      <ItalicsText {...italicsDefaultProps} />,
    );
    const textComponent = getByTestId(italicsDefaultProps.testID);
    expect(textComponent).toHaveStyleRule('font-style', 'italic');
    expect(textComponent).toHaveStyleRule('color', theme.vars.black);
    expect(textComponent).toMatchSnapshot();
  });

  it('renders with the color from prop', () => {
    const { getByTestId } = renderWithProps(
      <ItalicsText {...italicsColoredProps} />,
    );
    const textComponent = getByTestId(italicsColoredProps.testID);
    expect(textComponent).toHaveStyleRule('color', italicsColoredProps.color);
    expect(textComponent).toMatchSnapshot();
  });
});

describe('PlainText', () => {
  it('renders properly', () => {
    const { getByTestId } = renderWithProps(
      <PlainText {...plainDefaultProps} />,
    );
    const textComponent = getByTestId(plainDefaultProps.testID);
    expect(textComponent).toHaveStyleRule('font-style', 'normal');
    expect(textComponent).toHaveStyleRule('color', theme.vars.black);
    expect(textComponent).toMatchSnapshot();
  });

  it('renders with the color from prop', () => {
    const { getByTestId } = renderWithProps(
      <PlainText {...plainColoredProps} />,
    );
    const textComponent = getByTestId(plainColoredProps.testID);
    expect(textComponent).toHaveStyleRule('color', plainColoredProps.color);
    expect(textComponent).toMatchSnapshot();
  });
});
