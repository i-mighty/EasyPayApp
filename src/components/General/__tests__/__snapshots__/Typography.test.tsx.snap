// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItalicsText renders properly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontStyle": "italic",
      },
    ]
  }
  testID="ItalicsText"
/>
`;

exports[`ItalicsText renders with the color from prop 1`] = `
<Text
  color="#F89505"
  style={
    Array [
      Object {
        "color": "#F89505",
        "fontStyle": "italic",
      },
    ]
  }
  testID="ItalicsText"
/>
`;

exports[`PlainText renders properly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontStyle": "normal",
      },
    ]
  }
  testID="PlainText"
/>
`;

exports[`PlainText renders with the color from prop 1`] = `
<Text
  color="#F89505"
  style={
    Array [
      Object {
        "color": "#F89505",
        "fontStyle": "normal",
      },
    ]
  }
  testID="PlainText"
/>
`;

exports[`TitleText renders properly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontSize": 21,
      },
    ]
  }
  testID="TitleText"
/>
`;

exports[`TitleText renders with the color from prop 1`] = `
<Text
  color="#F89505"
  style={
    Array [
      Object {
        "color": "#F89505",
        "fontSize": 21,
      },
    ]
  }
  testID="TitleText"
/>
`;
